constraint-solver/src/solver.rs

```diff
@@ -156,6 +171,17 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display + Debug> Solver<T, V>
         !equivalences.is_empty()
     }
 
+    fn solve_with_backtracking(&mut self) -> Result<bool, Error> {
```

Comment:
At this point, it's not really backtracking any more, maybe 'exhaustive search'?